src/core/router/index.tsx

```diff
@@ -123,16 +123,22 @@ const Router = () => {
 
   const settingsQuery = useGetSettingsQuery(undefined, { skip: apikey === '' });
   const { theme } = settingsQuery.data?.WebUI_Settings ?? initialSettings.WebUI_Settings;
+  const bodyRef = useRef<HTMLDivElement>(null);
 
   useEffect(() => {
     document.body.className = `${apikey === '' ? globalThis.localStorage.getItem('theme') : (webuiPreviewTheme ?? theme)} theme-shoko-gray`;
-    setTimeout(() => {
-      document.getElementsByTagName('body')[0].style.display = 'initial';
+    const timeoutId = setTimeout(() => {
```

Comment:
Why the magic number and timeout? Is it possible to just have something like this?
```
useEffect(() => {
 if (bodyRef.current) {
        bodyRef.current.style.visibility = 'initial';
      }
}, []);
```

Reply:
We need the timeout to stop the flashing on the login page and 125 is the lowest I was able to go to achieve this while still being snappy. 

https://github.com/ShokoAnime/Shoko-WebUI/assets/9443295/d7a32fe1-d55f-4699-a1f4-52f30e8926ea